portkey_ai/api_resources/apis/chat_complete.py

```diff
@@ -20,182 +25,111 @@
 class ChatCompletion(APIResource):
     completions: Completions
 
-    def __init__(self, client: APIClient) -> None:
+    def __init__(self, client: Portkey) -> None:
         super().__init__(client)
         self.completions = Completions(client)
 
 
 class AsyncChatCompletion(AsyncAPIResource):
     completions: AsyncCompletions
 
-    def __init__(self, client: AsyncAPIClient) -> None:
+    def __init__(self, client: AsyncPortkey) -> None:
         super().__init__(client)
         self.completions = AsyncCompletions(client)
 
 
 class Completions(APIResource):
-    def __init__(self, client: APIClient) -> None:
+    def __init__(self, client: Portkey) -> None:
         super().__init__(client)
+        self.openai_client = client.openai_client
+
+    def stream_create(
+        self, model, messages, **kwargs
+    ) -> Union[ChatCompletions, Iterator[ChatCompletionChunk]]:
+        with self.openai_client.with_streaming_response.chat.completions.create(
+            model=model, messages=messages, **kwargs
+        ) as response:
+            for line in response.iter_lines():
+                json_string = line.replace("data: ", "")
+                json_string = json_string.strip().rstrip("\n")
+                if json_string == "":
+                    continue
+                elif json_string == "[DONE]":
+                    break
+                elif json_string != "":
+                    json_data = json.loads(json_string)
+                    json_data = ChatCompletionChunk(**json_data)
+                    yield json_data
+                else:
+                    return ""
+
+    def normal_create(self, model, messages, **kwargs) -> ChatCompletions:
+        response = self.openai_client.with_raw_response.chat.completions.create(
+            model=model, messages=messages, **kwargs
+        )
+        json_response = json.loads(response.text)
+        return ChatCompletions(**json_response)
```

Comment:
I think due to this, get_headers() is also not working because only json fields are getting passed to the model. Need to debug this further. But get_headers() is returning None. Please check this once.

Reply:
Done